burn-import/src/burn/node/constant.rs

```diff
@@ -1,72 +1,153 @@
+        match &self.value {
+            ConstantValue::Float32(_) => Type::Scalar(ScalarType {
+                name,
+                kind: ScalarKind::Float32,
+            }),
+            ConstantValue::Int64(_) => Type::Scalar(ScalarType {
+                name,
+                kind: ScalarKind::Int64,
+            }),
+            ConstantValue::Tensor(tensor_type, _) => Type::Tensor(tensor_type.clone()),
         }
     }
 }
 
-impl<PS: PrecisionSettings> NodeCodegen<PS> for ConstantNode {
+impl<PS: PrecisionSettings> NodeCodegen<PS> for ConstantNode<PS> {
     fn output_types(&self) -> Vec<Type> {
-        vec![Type::Other(&self.output_ty)]
+        vec![self.output.clone()]
     }
 
     fn input_types(&self) -> Vec<Type> {
         vec![]
     }
 
+    fn field_type(&self) -> Option<Type> {
+        match &self.value {
+            ConstantValue::Tensor(tensor_type, _) => Some(Type::Tensor(tensor_type.clone())),
+            _ => None,
+        }
+    }
+
+    fn field_init(&self, with_record: bool) -> Option<TokenStream> {
+        match &self.value {
+            ConstantValue::Tensor(tensor_type, _) => {
+                let ty = tensor_type.ty();
+                let name = Ident::new(self.name.as_ref(), Span::call_site());
+                let shape = tensor_type.clone().shape.unwrap().to_tokens();
+                let dim = tensor_type.clone().dim.to_tokens();
+
+                if with_record {
+                    Some(quote! {
+                        let #name = record.#name;
+                    })
+                } else {
+                    Some(quote! {
+                        let #name: #ty = Tensor::<B, #dim>::zeros(#shape);
+                    })
+                }
+            }
+            _ => None,
+        }
+    }
+
     fn forward(&self, _scope: &mut Scope, _node_position: usize) -> TokenStream {
         let name = Ident::new(self.name.as_ref(), Span::call_site());
-        let val = self.value.val_tokens();
-        let ty = self.value.ty_tokens();
+        let output = self.output.name();
 
-        quote! {
-            let #name: #ty = #val;
+        match &self.value {
+            ConstantValue::Tensor(_, _) => {
+                quote! {
+                    let #output = self.#name.clone();
+                }
+            }
+            _ => {
+                let val = self.value.val_tokens();
+                let ty = self.value.ty_tokens();
+
+                quote! {
+                    let #output: #ty = #val;
+                }
+            }
         }
     }
 
     fn into_node(self) -> Node<PS> {
         Node::Constant(self)
     }
+
+    fn field_serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        match &self.value {
+            ConstantValue::Tensor(_tensor_type, ds) => match ds {
+                TensorValue::Float(ds) => ds.serialize(serializer),
+                TensorValue::Int(ds) => ds.serialize(serializer),
+            },
```

Comment:
I guess if we update the tensor type to `Param<Tensor>` we can use the `Record` API like in the `linear node`.